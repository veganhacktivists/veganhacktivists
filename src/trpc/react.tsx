'use client';

import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { loggerLink, unstable_httpBatchStreamLink } from '@trpc/client';
import { createTRPCReact } from '@trpc/react-query';
import { type inferRouterInputs, type inferRouterOutputs } from '@trpc/server';
import { useState } from 'react';
import superjson from 'superjson';
import { Prisma } from '@prisma/client';

import type { AppRouter } from 'server/api/root';

const createQueryClient = () => new QueryClient();

let clientQueryClientSingleton: QueryClient | undefined = undefined;
const getQueryClient = () => {
  if (typeof window === 'undefined') {
    // Server: always make a new query client
    return createQueryClient();
  }
  // Browser: use singleton pattern to keep the same query client
  return (clientQueryClientSingleton ??= createQueryClient());
};

export const api = createTRPCReact<AppRouter>();

/**
 * Inference helper for inputs.
 *
 * @example type HelloInput = RouterInputs['example']['hello']
 */
export type RouterInputs = inferRouterInputs<AppRouter>;

/**
 * Inference helper for outputs.
 *
 * @example type HelloOutput = RouterOutputs['example']['hello']
 */
export type RouterOutputs = inferRouterOutputs<AppRouter>;

superjson.registerCustom<Prisma.Decimal, string>(
  {
    isApplicable: (v): v is Prisma.Decimal => Prisma.Decimal.isDecimal(v),
    serialize: (v) => v.toJSON(),
    deserialize: (v) => new Prisma.Decimal(v),
  },
  'decimal.js',
);

export const TRPCReactProvider = (props: { children: React.ReactNode }) => {
  const queryClient = getQueryClient();

  const [trpcClient] = useState(() =>
    api.createClient({
      links: [
        loggerLink({
          enabled: (op) =>
            process.env.NODE_ENV === 'development' ||
            (op.direction === 'down' && op.result instanceof Error),
        }),
        unstable_httpBatchStreamLink({
          transformer: superjson,
          url: getBaseUrl() + '/api/trpc',
          headers: () => {
            return { 'x-trpc-source': 'nextjs-react' };
          },
        }),
      ],
    }),
  );

  return (
    <QueryClientProvider client={queryClient}>
      <api.Provider client={trpcClient} queryClient={queryClient}>
        {props.children}
      </api.Provider>
    </QueryClientProvider>
  );
};

function getBaseUrl() {
  if (typeof window !== 'undefined') return window.location.origin;
  if (process.env.URL) return `https://${process.env.URL}`;
  return `http://localhost:${process.env.PORT ?? 3000}`;
}
