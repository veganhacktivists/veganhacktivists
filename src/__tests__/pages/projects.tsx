import React from 'react';
import { render } from '@testing-library/react';
import Projects from '../../pages/projects';

it('should render correctly', () => {
  const { asFragment } = render(<Projects />);
  expect(asFragment()).toMatchSnapshot();
});
