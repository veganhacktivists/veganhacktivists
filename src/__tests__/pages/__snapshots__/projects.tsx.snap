// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="relative md:-top-20 md:-mb-20 p-10 md:p-32 bg-scroll bg-cover bg-no-repeat flex justify-start bg-center"
    >
      <div
        style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
      >
        <img
          alt=""
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=%2Ffoo.png&w=3840&q=75"
          srcset="/_next/image?url=%2Ffoo.png&w=640&q=75 640w, /_next/image?url=%2Ffoo.png&w=750&q=75 750w, /_next/image?url=%2Ffoo.png&w=828&q=75 828w, /_next/image?url=%2Ffoo.png&w=1080&q=75 1080w, /_next/image?url=%2Ffoo.png&w=1200&q=75 1200w, /_next/image?url=%2Ffoo.png&w=1920&q=75 1920w, /_next/image?url=%2Ffoo.png&w=2048&q=75 2048w, /_next/image?url=%2Ffoo.png&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
        />
      </div>
      <div
        class="flex flex-col justify-center w-1/2 z-10"
      >
        <div>
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <noscript />
            <img
              alt="Building projects with impact"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </div>
      </div>
      <div
        class="absolute inset-0 overflow-hidden"
      >
        <svg
          class="text-white overflow-visible absolute top-0 right-0 transform translate-x-1/4 -translate-y-2/4 lg:-translate-y-3/4
    "
          height="60vw"
          width="60vw"
        >
          <circle
            class="stroke-current"
            cx="30vw"
            cy="30vw"
            fill-opacity="0"
            opacity="0.1"
            r="30vw"
            stroke-width="20px"
          />
        </svg>
        <svg
          class="text-white overflow-visible absolute bottom-0 left-0 transform -translate-x-1/4 translate-y-2/4 lg:translate-y-3/4
    "
          height="68vw"
          width="68vw"
        >
          <circle
            class="stroke-current"
            cx="34vw"
            cy="34vw"
            fill-opacity="0"
            opacity="0.5"
            r="34vw"
            stroke-width="20px"
          />
        </svg>
      </div>
    </div>
    <div
      class="relative overflow-visible hidden md:block"
    >
      <div
        class="absolute left-0 bottom-0 bg-white w-16 h-16 opacity-100"
      />
      <div
        class="absolute left-0 top-0 bg-orange w-16 h-16 opacity-100"
      />
      <div
        class="absolute left-16 bottom-0 bg-magenta w-32 h-32 opacity-100"
      />
      <div
        class="absolute right-0 -top-16 bg-yellow w-32 h-32 opacity-100"
      />
      <div
        class="absolute right-32 bottom-16 bg-yellow_orange w-16 h-16 opacity-100"
      />
      <div
        class="absolute right-32 bottom-0 bg-white w-16 h-16 opacity-100"
      />
    </div>
    <div
      class="text-grey content-center mx-auto my-32 md:w-1/2 drop-shadow-2xl text-2xl"
    >
      <p
        class="mb-16"
      >
        <span
          class="font-italic text-3xl"
        >
          Our 
        </span>
        <b
          class="font-mono text-5xl uppercase"
        >
          projects
        </b>
      </p>
      <p>
        We're constantly working on new projects every month, whether they're ideas of our own or supporting organizations and activists that reach out to us. Below is a list of our work, and which of our teams worked on it.
      </p>
    </div>
    <div
      class="flex flex-col space-y-20 items-center mx-auto drop-shadow-2xl text-2xl pb-20"
    >
      <p
        class="space-x-4"
      >
        <button
          type="submit"
        >
          <div
            class="border-2 border-gray-200 border-opacity-50 p-3 bg-gray text-white w-40 uppercase"
          >
            View all
          </div>
        </button>
        <button
          type="submit"
        >
          <div
            class="border-2 border-gray-200 border-opacity-50 p-3 w-40"
          >
            2021
          </div>
        </button>
        <button
          type="submit"
        >
          <div
            class="border-2 border-gray-200 border-opacity-50 p-3 w-40"
          >
            2020
          </div>
        </button>
        <button
          type="submit"
        >
          <div
            class="border-2 border-gray-200 border-opacity-50 p-3 w-40"
          >
            2019
          </div>
        </button>
        <button
          type="submit"
        >
          <div
            class="border-2 border-gray-200 border-opacity-50 p-3 w-40"
          >
            2018
          </div>
        </button>
      </p>
      <div
        class="flex space-x-8"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Activist Hub"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
        <div
          class="text-left"
          style="width: 700px;"
        >
          <h1
            class="text-4xl font-bold"
          >
            Activist Hub
          </h1>
          <p
            class="my-3"
          >
            <span
              class="font-bold text-grey"
            >
              APR 2021
            </span>
             - 
            <span
              class="uppercase font-bold"
            >
              Team Watermelon
            </span>
          </p>
          <p>
            The days of wondering whether or not our activism is effective are over. Activist Hub is the world's first street outreach dashboard that provides volunteers with the ability to monitor and understand their effectiveness through real data and analytics (DnA). Personal and group results will help everyone interested in making informed and strategic decisions for the animals!
          </p>
          <div
            class="mt-10 w-min"
          >
            <div
              class="p-3 px-4 py-2 text-2xl border-l-8 bg-w-x2 ease-out duration-1000 cursor-pointer hover:shadow-fill-green bg-grey-dark border-green text-white font-mono"
              href="https://activisthub.org"
            >
              ActivistHub.org
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex space-x-8"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Sehati Sanctuary"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
        <div
          class="text-left"
          style="width: 700px;"
        >
          <h1
            class="text-4xl font-bold"
          >
            Sehati Sanctuary
          </h1>
          <p
            class="my-3"
          >
            <span
              class="font-bold text-grey"
            >
              MAR 2021
            </span>
             - 
            <span
              class="uppercase font-bold"
            >
              Team Eggplant
            </span>
          </p>
          <p>
            Sehati Animal Sanctuary is the first farmed animal sanctuary in Indonesia which provides a loving haven to animals rescued from needless exploitation, violence, and slaughter. The sanctuary supports human-animal bonding and serves to educate the public about animal rights in order to reduce worldwide animal suffering.
          </p>
          <div
            class="mt-10 w-min"
          >
            <div
              class="p-3 px-4 py-2 text-2xl border-l-8 bg-w-x2 ease-out duration-1000 cursor-pointer hover:shadow-fill-green bg-grey-dark border-green text-white font-mono"
              href="https://sehatisanctuary.org"
            >
              SehatiSanctuary.org
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="relative overflow-visible hidden md:block"
    >
      <div
        class="absolute left-0 top-0 bg-white w-16 h-16 opacity-100"
      />
      <div
        class="absolute left-0 bottom-0 bg-gray-lighter w-16 h-16 opacity-100"
      />
      <div
        class="absolute right-0 bottom-0 bg-gray-lighter w-16 h-16 opacity-100"
      />
    </div>
    <div
      class="bg-black"
    >
      <div
        class="content-center mx-auto md:w-1/2 drop-shadow-2xl text-2xl pb-12"
      >
        <p
          class="mb-12 text-grey-dark pt-16"
        >
          <span
            class="text-5xl font-mono text-white"
          >
            JOIN OUR TEAM
          </span>
        </p>
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Compassion, Creativity, Code"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
        <p
          class="pb-5 mt-16 text-white"
        >
          Are you a developer, designer, writer, or other creative professional wanting to use skills for the animals? We're always looking for more vegan activist volunteers to join our team! If you're interested and have the time to spare, click to see our positions.
        </p>
        <div
          class="flex justify-center flex-wrap"
        >
          <a
            href="/team"
          >
            <div
              class="p-3 px-4 py-2 text-2xl border-l-8 bg-w-x2 ease-out duration-1000 cursor-pointer hover:shadow-fill-green text-grey-dark border-green bg-w-x2 bg-white m-5 font-mono text-sm"
            >
              Meet the Team
            </div>
          </a>
          <a
            href="/join"
          >
            <div
              class="p-3 px-4 py-2 text-2xl border-l-8 bg-w-x2 ease-out duration-1000 cursor-pointer hover:shadow-fill-green text-grey-dark border-green bg-w-x2 bg-white m-5 font-mono text-sm"
            >
              Apply to Join
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
